opensearch/Dockerfile

```diff
@@ -1,7 +1,30 @@
 # Repo name: arynai/sycamore-opensearch
 
 ARG TAG=2.12.0
-FROM opensearchproject/opensearch:$TAG
+
+FROM gradle:jdk17 as build_ml_commons
+
+WORKDIR /build_ml
+RUN git clone https://github.com/opensearch-project/ml-commons.git
+WORKDIR /build_ml/ml-commons
+RUN git checkout -b 2.12 origin/2.12
+RUN sed -i 's#if (os.macOsX) {#if (true) {#' ml-algorithms/build.gradle
+RUN sed -i 's#throw new OpenSearchException("GenerativeQAResponseProcessor failed in precessing response");#log.error(e); throw new OpenSearchException(e);#'\
+    search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessor.java
+RUN ./gradlew assemble -DoperatingSystem=macOsX
+
+FROM opensearchproject/opensearch:$TAG as install_ml_commons
+
+USER opensearch
+COPY --chown=opensearch:opensearch --from=build_ml_commons \
+    /build_ml/ml-commons/plugin/build/distributions/opensearch-ml-2.12.0.0-SNAPSHOT.zip /tmp/
+RUN /usr/share/opensearch/bin/opensearch-plugin remove opensearch-skills
```

Comment:
Why are we removing opensearch-skills and then re-installing it?

Reply:
it extends ml commons, so the opensearch-plugin tool won't let me remove ml-commons without first removing skills

Reply:
optional: add a comment to explain.